### PR TITLE
feat: add preflight for network connectivity (aka firewall preflight)

### PR DESCRIPTION
**Pull request by ricardomaraschini**
#### What this PR does / why we need it:

Checks if communication between pods and services CIDRs is allowed on the node before installing.

#### Does this PR require a release note?

```release-note
Added preflight for network communication.
```

### CODE REVIEW

File: pkg/cmd/network.go [hunk 1]
```diff
@@ -16,12 +17,24 @@ func withSubnetCIDRFlags(flags []cli.Flag) []cli.Flag {
...
+				return nil
+			},
 		},
 		&cli.StringFlag{
 			Name:   "service-cidr",
 			Usage:  "IP address range for Services",
 			Value:  k0sv1beta1.DefaultNetwork().ServiceCIDR,
 			Hidden: true,
+			Action: func(c *cli.Context, addr string) error {
+				if _, _, err := net.ParseCIDR(addr); err != nil {
```
Comment (laverya) [inline pkg/cmd/network.go:33]: should these use the same netutils code as the cidr flag?
maybe with a /17 requirement instead of /16

Comment (ricardomaraschini) [reply pkg/cmd/network.go:33]: I don't think we want to restrict this too much. I am worried with people that may be already using this with a `/24` or something. As this is now a hidden flag we may consider printing a warning here but I would argue this should be part of a different issue.

**Review by laverya — APPROVED** [review]

**Review by laverya — APPROVED** [review]

File: pkg/preflights/host-preflight.yaml [hunk 2]
```diff
@@ -880,3 +884,10 @@ spec:
           - pass:
               when: 'net.ipv4.ip_forward > 0'
               message: "IP forwarding is enabled."
+    - networkNamespaceConnectivity:
+        collectorName: check-network-namespace-connectivity
+        outcomes:
+        - pass:
+            message: Communication between {{ "{{ .FromNamespace }}" }} and {{ "{{ .ToNamespace }}" }} is working
+        - fail:
+            message: Network communication between the pods network CIDR {{ "{{ .FromNamespace }}" }} and the services network CIDR {{ "{{ .ToNamespace }}" }} is currently disrupted on this node. This issue may prevent pods from properly interacting with services. To resolve this, please review and adjust the firewall configuration on this node to ensure it allows traffic between these network CIDRs.
```
[the remark this answers is not in the document]
Comment (ricardomaraschini) [reply pkg/preflights/host-preflight.yaml:893]: Oh gosh, I wonder how I made this far without noticing this! Thanks a lot. This is wrong in troubleshoot and should be called FromCIDR and ToCIDR there instead. Can we get this in and then I address this a separated PR ?

[the remark this answers is not in the document]
Comment (laverya) [reply pkg/preflights/host-preflight.yaml:893]: oh it's from troubleshoot itself

**Review by laverya — APPROVED** [review]

**Comment by laverya** [comment]
Tested and working (Ubuntu 24.04, default UFW enabled)
```
✗  1 host preflight failed

 •  Network communication between the pods network CIDR 10.244.0.0/17 and the services network CIDR 10.244.128.0/17 is currently disrupted on this
    node. This issue may prevent pods from properly interacting with services. To resolve this, please review and adjust the firewall configuration
    on this node to ensure it allows traffic between these network CIDRs.
```

**Pull request merged**